src/engine/source/indexerconnector/test/unit/serverSelector_test.cpp

```diff
@@ -129,31 +111,67 @@ TEST_F(ServerSelectorTest, TestGetNextBeforeHealthCheck)
  */
 TEST_F(ServerSelectorTest, TestGetNextBeforeAndAfterHealthCheck)
 {
-    const auto hostGreenServer {m_servers.at(0)};
-    const auto hostRedServer {m_servers.at(1)};
+    // Set up the expectations for the MockHTTPRequest
+    EXPECT_CALL(*spHTTPRequest, get(::testing::_, ::testing::_, ::testing::_))
+        .WillRepeatedly(::testing::Invoke(
+            [](RequestParameters requestParameters,
+               PostRequestParameters postRequestParameters,
+               ConfigurationParameters /*configurationParameters*/)
+            {
+                const auto& url = requestParameters.url.url(); // Convert URL to string if needed
+                std::string response;
+
+                if (url == GREEN_SERVER + "/_cat/health?v")
+                {
+                    // Simulate a successful response for the green server
+                    response = "epoch\ttimestamp\tcluster\tstatus\tnode.total\tnode.data\tdiscovered_cluster_"
+                               "manager\tshards\tpri\trelo\tinit\tunassign\tpending_tasks\tmax_task_wait_time\tactive_"
+                               "shards_percent\n1725296432\t17:00:32\twazuh-"
+                               "cluster\tgreen\t1\t1\ttrue\t47\t47\t0\t0\t0\t0\t-\t100.0%\n";
+                    postRequestParameters.onSuccess(response.c_str());
```

Comment:
We are repeating this responses in all tests, consider using a global variable inside a namespace

Reply:
I moved outside. I think a namespace it's not necessary, but let me know if you want, I add it